src/HappyConnOpener.cc

```diff
@@ -717,11 +720,14 @@ HappyConnOpener::checkForNewConnection()
     // open a new prime and/or a new spare connection if needed
     if (!destinations->empty()) {
         if (!currentPeer) {
-            currentPeer = destinations->extractFront();
+            auto newPrime = destinations->extractFront();
+            currentPeer = newPrime;
             Must(currentPeer);
             debugs(17, 7, "new peer " << *currentPeer);
             primeStart = current_dtime;
-            startConnecting(prime, currentPeer);
+            startConnecting(prime, newPrime);
+            // XXX: if reuseOldConnection() in startConnecting() above succeeds,
+            // then we should not get here, and Must(prime) below will fail.
             maybeGivePrimeItsChance();
             Must(prime); // entering state #1.1
```

Comment:
I speculate that this bug has no significant runtime effect (in most cases) because a successful startConnecting() sends the persistent connection to the requestor. The subsequent and immediate HappyConnOpener death (due to the Must failure) cannot hurt the requestor (or the already "done" HappyConnOpener job) in any way.

Reply:
again "no significant effect" == TODO,  not XXX.

Reply:
IHMO it would be strange to mark bugs (even rare ones) with TODOs.

Reply:
@eduard-bagdasaryan, do you mean leave this known issue unmarked? or fix it in this PR?

Reply:
No, I meant that a known/discovered bug should be better marked with XXX rather than with TODO. I do not think we should fix this unrelated problem in this PR.

Reply:
I agree with Eduard -- most bugs should be marked with an XXX.

Unfortunately, it is difficult to define "bug" precisely, especially for folks that prefer literal interpretation of definitions. I tried to avoid that trap when sketching XXX/TODO differences in 9837567dd913854a4deddcc49043bfd7631ab63f by using the word "problem" for both. Clearly, that opened another attack vector!

Reply:
I am judging the selection between them based on severity alone, per that ad-hoc policy description.

Eduard is saying he thinks it is serious -> XXX

Alex is saying it has "no significant effect" aka not serious at all.

What is anyone else to think?   The comment indicates serious (XXX) and code has a Must() which implies there is a handler already to cope with it cleanly (so, serious bug? or already managed?).

Reply:
> Eduard is saying he thinks it is serious -> XXX

Eduard did _not_ say that AFAICT. He said it is a bug. I agreed. We both think bugs should be marked with an XXX.

> Alex is saying it has "no significant effect" aka not serious at all.

I am not saying just "no significant effect". I said "no significant runtime effect (in most cases)". There are enough loopholes in that phrase to accommodate for serious effects.

Besides, the "problem, often a serious one" does not imply that all XXXs are serious problems.

If replacing the letters `XXX` with `TODO` would allow us to resolve this thread, we should replace it: The time wasted here has already negated any benefit of perfecting this comment!

Reply:
Done at 31c8392 to facilitate the PR progress.

Reply:
BTW, I have a crazy theory where the disagreement about the policy [sketch](https://github.com/squid-cache/squid/commit/9837567dd913854a4deddcc49043bfd7631ab63f) may be coming from.

> XXX: a problem, often a serious one

The word "often" in the above phrase applies to the entire set of XXXs in Squid, not every single XXX. In other words, "many XXXs mark serious problems" not "every XXX marks a problem that is often serious". Better wording to remove this alleged ambiguity is welcomed.

> a Must() which implies there is a handler already to cope with it cleanly

FYI: Squid Must()s do not carry such implications.

----

Please do not misinterpret this comment as a resurrection of this change request thread. I (still) do not object to the wrong label used for marking this bug. I am just desperately trying to avoid wasting so much time on _future_ markings.